### ISSUE TITLE
DetNAS (COCO-FPN) release

### ISSUE THREAD

**Issue by zvadaszi** [megvii-model/ShuffleNet-Series#7]
I'm integrating DetNAS into mmdetection library and eagerly waiting for your detector optimized models, especially DetNAS (COCO-FPN) ones.
The provided ClsNASNet_medium already shows improvement over ResNet50 as a backbone in a Cascade R-CNN detector.

When do you plan to release the detector optimized models?
Thank you!

**Comment by nmaac** [comment]
Hi @zvadaszi , the DetNAS models we released can be directly used as detector backbones. We don't use open source frameworks in our research works, but we are also planning to develop PyTorch version detector framework and release models for public use.

**Comment by zvadaszi** [comment]
Hi @nmaac , yes I'm currently evaluating the DetNAS models you released.
In the 5.3 part of the DetNAS paper the topmost network, similar to the ones you released - with large 7x7 kernels in Stage 2 and 3, is mentioned as ClsNasNet while they argue that for detection the other architectures, DetNAS (COCO-FPN) and DetNAS (COCO-RetinaNet), perform much better.
I thought you plan to release pretrained models for these architectures as well.

**Comment by nmaac** [comment]
Hi @zvadaszi , the 1.3G and 3.8G models we released are already COCO-FPN, for 300M we have added all the COCO/VOC-FPN/RetinaNet backbone models.

**Comment by nmaac** [comment]
Hi @zvadaszi , DetNAS only search backbones for detector. 

Please refer to Fig. 2 in the paper, the models we released are  (b) DetNAS (COCO-FPN) and (c) DetNAS (COCO-RetinaNet), which are backbones searched for COCO-FPN/RetinaNet architectures. We didn't release Fig. 2(a) ClsNasNet which is searched on ImageNet.

Specifically, 
- for 300M we have released COCO/VOC-FPN/RetinaNet backbone models;
- for 1.3G and 3.8G, we have released COCO-FPN backbone models.

**Comment by zvadaszi** [comment]
Hi @nmaac , I was mislead by the explanation from 5.3 and presumed that the large 40 blocks DetNAS (COCO-FPN) architecture should resemble the 20 blocks ones, which is not the case.

Thank you for your clarification and for this great contribution.

**Comment by ggjy** [comment]
hi @zvadaszi, What's the data augmentation you use for COCO? When I directly used COCO_300M as detector backbones, (data: img mean[0, 0, 0], var[1, 1, 1] bgr) I only get 32% mAP in 1X setting.

**Comment by zvadaszi** [comment]
hi @ggjy , in mmdetection I'm using these values for image normalization, seems to work fine:
```
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[1., 1., 1.], to_rgb=True)
```
I usually do a quick test for ImageNet models: I classify a Labrador dog with them. If I normalize the image correctly the prediction should be correct with high probability, otherwise it's crap.

**Comment by ggjy** [comment]
@zvadaszi Thanks! I will try it now. But it seems that in this repo, they use function ToBGRTensor(). When I test the ImageNet dataset myself, ues the pretrained model in OneDrive, I find that I need to **not** normalize the input image, otherwise it gets 0 acc.

**Comment by ggjy** [comment]
Do you change the BN to SyncBN?

**Comment by zvadaszi** [comment]
> Do you change the BN to SyncBN?
I've used Group Norm for the detection layers and left the BN for the backbone.
I removed the last conv layer (1024 from the COCO_FPN_1.3G) and trained only conv 4 and 5 (with 480 and 960 filters). I tried to train with conv 3 as well but overfitted (for our dataset).
